File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;
class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = {0, 0};
        String prev = "";
        HashSet<String> used = new HashSet<String>();
        int turn = 1;
        
        for (int i = 0; i < words.length; i++) {
            String current = words[i];
            if (i != 0 && current.charAt(0) != prev.charAt(prev.length() - 1)) {
                answer[0] = turn;
                answer[1] = (i / n) + 1;
                break;
            } else if (used.contains(current)) {
                answer[0] = turn;
                answer[1] = (i / n) + 1;
                break;
            } else {
                used.add(current);
                prev = current;
                turn = (turn == n) ? 1 : turn + 1;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = []
    usedWord = set()
    usedWord.add(words[0])
    for i in range(1, len(words)):
        if words[i][0] == words[i-1][-1] and words[i] not in usedWord:
            usedWord.add(words[i])
        else:
            return [(i % n) + 1, (i // n) + 1]

    return [0, 0]
